Libraries/WebSocket/RCTSRWebSocket.m

```diff
@@ -245,7 +245,7 @@ + (void)initialize;
   CRLFCRLF = [[NSData alloc] initWithBytes:"\r\n\r\n" length:4];
 }
 
-- (instancetype)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray<NSString *> *)protocols options:(NSDictionary<NSString *, NSString *> *)options
+- (instancetype)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray<NSString *> *)protocols headers:(NSDictionary<NSString *, NSString *> *)headers
```

Comment:
@nicklockwood That makes sense but it appears there are two other places that headers are being set.
1. https://github.com/facebook/react-native/blob/debcac59d06d527c3a6a07ef9561fe4041022ff4/Libraries/WebSocket/RCTSRWebSocket.m#L478-L489
2. https://github.com/facebook/react-native/blob/debcac59d06d527c3a6a07ef9561fe4041022ff4/Libraries/WebSocket/RCTSRWebSocket.m#L298

So im not sure the best way to handle this.